src/jshint.js

```diff
@@ -1419,12 +1425,23 @@ var JSHINT = (function() {
     var x = symbol(s, 150);
 
     x.led = function(left) {
+      // this = suffix e.g. "++" punctuator
+      // left = symbol operated e.g. "a" identifier or "a.b" punctuator
+
       if (state.option.plusplus) {
         warning("W016", this, this.id);
       } else if ((!left.identifier || isReserved(left)) && left.id !== "." && left.id !== "[") {
         warning("W017", this);
       }
 
+      // detect modification of a const.
+      // in the case of a.b, left will be the "." punctuator
```

Comment:
It’s called for a number of prefixes, preinc, predec, unary minus, unary plus. The unary ops are totally fine because they’re not assignment ops.

> On Apr 26, 2015, at 8:44 AM, Luke Page <notifications@github.com> wrote:
> 
> In src/jshint.js <https://github.com/jshint/jshint/pull/2326#discussion_r29109438>:
> 
> >        if (state.option.plusplus) {
> >          warning("W016", this, this.id);
> >        } else if ((!left.identifier || isReserved(left)) && left.id !== "." && left.id !== "[") {
> >          warning("W017", this);
> >        }
> >  
> > +      // detect modification of a const.
> > +      // in the case of a.b, left will be the "." punctuator
> its in a function that is only called for ++ and -- so what should I say?
> I've changed for // detect increment/decrement of a const ?
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/jshint/jshint/pull/2326/files#r29109438>.
>